<launch>
    <!-- Set parameters for the dummy publishers -->
    <param name="dummy_elevation" type="double" value="7.8" />
    <param name="dummy_bearing" type="double" value="358" />

    <!-- Set parameters for geolocation scripts -->
    <param name="marker_array_topic" type="string" value="/marker_array" />

    <!-- Launch dummy publishers to populate topics -->
    <!-- node name="dummy_point" pkg="geolocation" type="dummy_pt_pub" /-->
    <node name="dummy_bearing" pkg="geolocation" type="dummy_bearing_pub" />
    <node name="dummy_elevation" pkg="geolocation" type="dummy_elv_pub" />
    <node name="dummy_gnss" pkg="geolocation" type="dummy_gnss_pub" />
    <node name="dummy_marker_array" pkg="geolocation" type="dummy_markers_pub" />

    <!-- Launch pose publishers and tf2 broadcasters for pose and transforms -->
    <node name="pt_to_pose" pkg="geolocation" type="pt_to_pose_pub" />
    <node name="track_tf2" pkg="geolocation" type="track_tf2_broadcaster" />
    <node name="map_tf2" pkg="geolocation" type="map_tf2_broadcaster" />

    <!-- Launch geolocation script -->
    <node name="geolocation" pkg="geolocation" type="geo_publisher.py" args="0" output="screen" />

    <!-- Launch rviz with the preconfiguration -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find geolocation)/launch/demo.rviz"/>

    <!-- Launch ROS Bridge -->
    <node name="rosbridge" pkg="rosbridge_server" type="rosbridge_websocket" output="screen" />
        <param name="port" value="9090"/>

</launch>